addon/utils/assign.ts

```diff
@@ -1,6 +1,23 @@
+let getOwnPropertyDescriptors: (obj: { [x: string]: any }) => { [x: string]: PropertyDescriptor };
+
+if (Object.getOwnPropertyDescriptors !== undefined) {
+  getOwnPropertyDescriptors = Object.getOwnPropertyDescriptors;
+} else {
+  getOwnPropertyDescriptors = function(obj: object) {
```

Comment:
`obj: Record<string, any>`

I think we need this so accessing `descriptors[key]` doesn't fail with an index signature error.

Reply:
thanks, fixed